client/components/data/query-theme/use-query-single-theme.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'calypso/state';
+import { requestSingleTheme } from 'calypso/state/themes/actions/request-theme';
+import { isRequestingTheme } from 'calypso/state/themes/selectors';
+import { CalypsoDispatch } from 'calypso/state/types';
+import { AppState } from 'calypso/types';
+
+export type SourceType = 'wpcom' | 'wporg' | 'jetpack';
+
+const request =
+	( siteId: number | null, themeId: string, sourceType: SourceType ) =>
+	( dispatch: CalypsoDispatch, getState: AppState ) => {
+		if ( ! isRequestingTheme( getState(), siteId as number, themeId ) ) {
+			dispatch( requestSingleTheme( themeId, siteId, sourceType ) );
+		}
+	};
+
+export function useQuerySingleTheme(
```

Comment:
I think that we could simplify this file a bit, what do you think of this approach?

````
import { useDispatch, useSelector } from 'calypso/state';
import { requestSingleTheme } from 'calypso/state/themes/actions/request-theme';
import { isRequestingTheme } from 'calypso/state/themes/selectors';

export type SourceType = 'wpcom' | 'wporg' | 'jetpack';

export function useQuerySingleTheme( siteId: number, themeId: string, sourceType: SourceType ) {
	const dispatch = useDispatch();
	const isRequesting = useSelector( ( state ) => isRequestingTheme( state, siteId, themeId ) );

	if ( ! isRequesting ) {
		dispatch( requestSingleTheme( themeId, siteId, sourceType ) );
	}
}
````

I think that code should be equivalent, the sourceType and themeId props are mandatory so we wouldn't need to check them unless I'm missing something.